SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+		};
+
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckRequiresHandover(this, true));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_default_strat_WHEN_who_should_have_auth_called_THEN_return_1)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy({}, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnDefaultPawnAtLocation(TEXT("Actor1"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckWhoShouldHaveAuthority(this, "Actor1", 1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_two_actors_of_same_type_at_same_position_WHEN_who_should_have_auth_called_THEN_return_same_for_both)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+
+		std::vector<StrategyConfig> ComponentStrategyConfig
+		{
+			{UTwoByFourLBGridStrategy::StaticClass(), {ADefaultPawn::StaticClass()}},
+			{UTwoByFourLBGridStrategy::StaticClass(), {ASpecialPawn::StaticClass()}}
+		};
+
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnDefaultPawnAtLocation(TEXT("Actor1"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnDefaultPawnAtLocation(TEXT("Actor2"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnSpecialPawnAtLocation(TEXT("Actor3"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnSpecialPawnAtLocation(TEXT("Actor4"), FVector(0, 0, 0)));
+
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckActorsHaveSameAuth(this, TEXT("Actor1"), TEXT("Actor2")));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckActorsHaveSameAuth(this, TEXT("Actor3"), TEXT("Actor4")));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_default_and_2_2x4_grids_WHEN_should_have_auth_called_THEN_return_true_iff_who_should_have_auth_returns_local_worker_id)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+
+		std::vector<StrategyConfig> ComponentStrategyConfig{
+			{UTwoByFourLBGridStrategy::StaticClass(), {ADefaultPawn::StaticClass()}},
+			{UTwoByFourLBGridStrategy::StaticClass(), {ASpecialPawn::StaticClass()}}
+		};
+
+		int num_virtual_workers = 17;
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, num_virtual_workers));
+
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnDefaultPawnAtLocation(TEXT("Actor1"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnSpecialPawnAtLocation(TEXT("Actor2"), FVector(0, 0, 0)));
+		for (int i = 1; i <= num_virtual_workers; ++i) {
```

Comment:
This is probably not a thing we want to test. Ideally, the LBStrat would return an error if SetLocalVirtualWorkerId is called twice.   (Actually, if you want to add that error, that would be great.)

Reply:
Added the error and changed the test